pose_follower/src/pose_follower.cpp

```diff
@@ -43,68 +43,60 @@ PLUGINLIB_EXPORT_CLASS(pose_follower::PoseFollower, nav_core::BaseLocalPlanner)
 namespace pose_follower {
   PoseFollower::PoseFollower(): tf_(NULL), costmap_ros_(NULL) {}
 
+  PoseFollower::~PoseFollower() {
+    if (dsrv_)
+      delete dsrv_;
+  }
+
   void PoseFollower::initialize(std::string name, tf2_ros::Buffer* tf, costmap_2d::Costmap2DROS* costmap_ros){
     tf_ = tf;
     costmap_ros_ = costmap_ros;
     current_waypoint_ = 0;
     goal_reached_time_ = ros::Time::now();
     ros::NodeHandle node_private("~/" + name);
 
-    collision_planner_.initialize(name, tf_, costmap_ros_);
+    collision_planner_.initialize(name + "/collision_planner", tf_, costmap_ros_);
 
     node_private.param("k_trans", K_trans_, 2.0);
     node_private.param("k_rot", K_rot_, 2.0);
```

Comment:
Is there a reason why you didn't also make `k_trans`, `k_rot` and `holonomic` into dynamic parameters?

Reply:
`k_trans` and `k_rot` can be added as well. I like to considered them as a constant scaling factors since I am not sure how intuitive is to tinker with them dynamically. As for the `holonomic` parameter, I think that it should be configured only once (in a launch file) since it depends on the type of the robot used. If you are using a differential drive mobile base (or vice versa) then there is no point in changing this parameter on the run.

Reply:
Agreed. I've just turned `k_trans` and `k_rot` into dynamic parameters and left `holonomic` alone.